jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -130,7 +130,12 @@
 public class ServletContextHandler extends ContextHandler
 {
     private static final Logger LOG = LoggerFactory.getLogger(ServletContextHandler.class);
-    public static final Environment __environment = Environment.ensure("ee10");
+    public static final Environment ENVIRONMENT = Environment.ensure("ee10");
+    /**
+     * @deprecated Use {@link ServletContextHandler#ENVIRONMENT} instead. will be removed in Jetty 12.1.0.
```

Comment:
Remove the part related to when it will be removed.

Reply:
I disagree, this is super common in use on open source java projects (spring-boot, spring-framework, dropwizard, tomcat, etc) and benefits developers.
Our own codebase already has many such usages before this PR.

Reply:
@joakime then we should remove them all.

Spring has a policy (2 minor versions).

We do not know if we ever will have a 12.1.0, maybe we jump to 13 immediately due to the move to Java 21 required by JEE 11.

Or maybe we will have a policy similar to Spring.

Or maybe we retain it for sponsored reasons.

So the comment about 12.1.0 is wrong, in all cases, and must be removed.

Reply:
I'm on the fence here.   I like seeing the deprecated for removal.   I also don't mind being given info that will let me know I can keep using it (e.g. knowing it will always be there in 12.0.x).       But I also can see that naming a specific future version is not always correct (although we have frequently done it before and survived!).

So perhaps just say it is deprecated for removal "after 12.0.x", and that could be 12.1.0 or 13.0.0

Reply:
> So the comment about 12.1.0 is wrong, in all cases, and must be removed.

I disagree with the "must be removed" part of your reply.
but I'm totally on board with having a minimum 2 minor releases policy.
That would change these references to say "12.2.0" instead, which means even if 13.0.0 gets released, they are removed, as 13.0.0 is after 12.2.0.

Reply:
@joakime then the policy is outlined elsewhere, e.g. in the docs.
Leave the code clean, without comments that are going to rot.

Reply:
The policy on deprecated doesn't exist in any documentation in Jetty.

Reply:
Sure, let's add it, then. But it is orthogonal to this issue.